RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitConvertGPU2CPUFormat.cc

```diff
@@ -0,0 +1,123 @@
+// framework
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h" 
+
+// algorithm specific
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CUDADataFormats/EcalRecHitSoA/interface/EcalUncalibratedRecHit_soa.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "RecoLocalCalo/EcalRecAlgos/interface/Common.h"
+
+#include <iostream>
+
+class EcalUncalibRecHitConvertGPU2CPUFormat
+    : public edm::stream::EDProducer<>
+{
+public:
+    explicit EcalUncalibRecHitConvertGPU2CPUFormat(edm::ParameterSet const& ps);
+    ~EcalUncalibRecHitConvertGPU2CPUFormat() override;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+    using GPURecHitType = ecal::UncalibratedRecHit<ecal::Tag::soa>;
+    void produce(edm::Event&, edm::EventSetup const&) override;
+
+private:
+    edm::EDGetTokenT<ecal::SoAUncalibratedRecHitCollection> recHitsGPUEB_;
```

Comment:
can you make these `const` ?

Reply:
sorry, i did not get what should be made const???

Reply:
```c++
const edm::EDGetTokenT<ecal::SoAUncalibratedRecHitCollection> recHitsGPUEB_;
const edm::EDGetTokenT<ecal::SoAUncalibratedRecHitCollection> recHitsGPUEE_;
const std::string recHitsLabelCPUEB_;
const std::string recHitsLabelCPUEE_;
```